nixopsvirtd/backends/libvirtd.py

```diff
@@ -200,27 +221,40 @@ def _prepare_storage_volume(self):
 
         self.logger.log("uploading disk image...")
         image_info = self._get_image_info(temp_disk_path)
-        self._vol = self._create_volume(image_info['virtual-size'], image_info['actual-size'])
+        self._vol = self._create_volume(image_info['virtual-size'], image_info['actual-size'], defn.image_dir)
         self._upload_volume(temp_disk_path, image_info['actual-size'])
 
     def _get_image_info(self, filename):
         output = self._logged_exec(["qemu-img", "info", "--output", "json", filename], capture_stdout=True)
         return json.loads(output)
 
-    def _create_volume(self, virtual_size, actual_size):
+    def _create_volume(self, virtual_size, actual_size, path=None):
+        # according to https://libvirt.org/formatstorage.html#StoragePoolTarget
+        # files should be created with rights depending on parent folder but 
+        # this doesn't seem true
+        # here I hardcode permission rights (BAD)
         xml = '''
         <volume>
           <name>{name}</name>
           <capacity>{virtual_size}</capacity>
           <allocation>{actual_size}</allocation>
           <target>
             <format type="qcow2"/>
+            <permissions>
+                <owner>1000</owner>
+                <group>100</group>
+                <mode>0744</mode>
+                <label>virt_image_t</label>
+            </permissions>
```

Comment:
I think this needs to be fixed before merge. Is there an upstream libvirt bug about the mismatch in docs and reality mentioned in the comment above?

Explicitly specifying owner, group and SELinux label will break many setups, including mine.

Reply:
it might have become necessary with the other PR merged today. I will look into it

Reply:
so if I understand the doc correctly, the pool is created with the permission of the folder but the volums are created by default with the user/group that libvirtd is running as. Seems like the libvirtd service is running as root without any nix option to change it so this looks a bit more complex than I have time for (for now) so feel free to look into it.